sdk/keyvault/keyvault-keys/src/lro/keyVaultKeyPoller.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { delay, RequestOptionsBase } from "@azure/core-http";
+import { Poller, PollOperation, PollOperationState } from "@azure/core-lro";
+import { KeyVaultClient } from "../generated/keyVaultClient";
+
+/**
+ * Common parameters to a Key Vault Key Poller.
+ */
+export interface KeyVaultKeyPollerOptions {
+  vaultUrl: string;
+  client: KeyVaultClient;
+  name: string;
+  requestOptions?: RequestOptionsBase;
+  intervalInMs?: number;
+  resumeFrom?: string;
+}
+
+/**
+ * An interface representing the state of a Key Vault Key Poller's operation.
+ */
+export interface KeyVaultKeyPollOperationState<TResult> extends PollOperationState<TResult> {
+  /**
+   * The name of the key.
+   */
+  name: string;
+}
+
+/**
+ * Common properties and methods of the Key Vault Key Pollers.
+ */
+export abstract class KeyVaultKeyPoller<TState, TResult> extends Poller<TState, TResult> {
+  /**
+   * Defines how much time the poller is going to wait before making a new request to the service.
+   * @memberof DeleteKeyPoller
+   */
+  public intervalInMs: number = 2000;
+
+  /**
+   * The method used by the poller to wait before attempting to update its operation.
+   * @memberof DeleteKeyPoller
+   */
+  async delay(): Promise<void> {
```

Comment:
Could we provide a default implementation of this method in `core-lro`? It already depends on `core-http`, and I'm having a hard time thinking of where anyone would use an implementation other than this in a package, but leaving it open to overrides would still be nice if someone wanted to instrument the delay for some purpose.

Reply:
The issue with this is that we didn't want core-lro to depend on core-http 🤔 and doing an independent delay function seems prone to errors. Would you be on board if I did a default delay method with a non-core-http delay?

Reply:
`core-lro` already depends on `core-http`.

```
⇒  < ../../core/core-lro/package.json | jq .dependencies
{
  "@azure/abort-controller": "^1.0.0",
  "@azure/core-http": "^1.2.0",
  "events": "^3.0.0",
  "tslib": "^2.0.0"
}
```

The true problem is that the base `Poller` doesn't contain a polling interval. If it had one, I think we could just use `setTimeout` on all targets by default? (CC @xirzec )

For cancellation we would just have to store the timer ID and handle it when the operation is cancelled.

Reply:
stopping the poller and cancelling the operation is different! but I see what you mean. I'll make an issue.

Reply:
I made an issue! https://github.com/Azure/azure-sdk-for-js/issues/12583

Reply:
Having some kind of default delay implementation makes sense to me.